chi/container.py

```diff
@@ -292,6 +326,36 @@ def associate_floating_ip(self, fip: str = None):
         """
         return associate_floating_ip(self.id, fip)
 
+    def detach_floating_ip(self, fip: str) -> None:
+        """
+        Detaches and deletes a floating IP from the container.
+
+        Args:
+            fip (str): The floating IP to detach.
+            delete (Optional[bool], optional): Whether to delete the floating IP after disassociation. Defaults to True.
+
+        Returns:
+            None
+        """
+        conn = connection(session=session())
+        floating_ip_obj = chi_network.get_floating_ip(fip)
+        conn.network.delete(floating_ip_obj["id"])
```

Comment:
It would be nice to be able to pass in a connection or session object, rather than using the global session. Particularly for CHI@Edge, it's common to want to configure floating IPs for say, chi@edge and KVM in one notebook.

Reply:
Yeah, ideally we would have a better paradigm for this overall.